StatusDialogManagerTest.testBug275867() flaky in Windows I-Builds

The test case `org.eclipse.ui.tests.statushandlers.StatusDialogManagerTest.testBug275867()` sporadically fails in I-Builds on Windows:

```java
Dialog is not centered correctly expected:<Point {254, 102}> but was:<Point {254, 103}>

java.lang.AssertionError: Dialog is not centered correctly expected:<Point {254, 102}> but was:<Point {254, 103}>
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failNotEquals(Assert.java:835)
at org.junit.Assert.assertEquals(Assert.java:120)
at org.eclipse.ui.tests.statushandlers.StatusDialogManagerTest.testBug275867(StatusDialogManagerTest.java:777)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

From the values, it might be some rounding issue related to HiDPI support / monitor-specific scaling.
As an initial thought if we cannot reproduce the fail locally or have any other clue: we might add debug output with the actual and expected values to the test, such that we know how the values differ in case the test succeeds. It would be strange if in case of a success the values are the exact same as we see in case of a failure (except for the off-by-one), as there should be no non-determinism in how values are calculated/rounded. That's why I would expect the values to be completely different in case of a success, which could be verified by debug output.

--- a/tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/statushandlers/StatusDialogManagerTest.java
+++ b/tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/statushandlers/StatusDialogManagerTest.java
@@ -774,7 +774,16 @@ public class StatusDialogManagerTest {
 		wsdm.addStatusAdapter(statusAdapter, false);
 		assertNotNull(StatusDialogUtil.getStatusShell());
 		Shell shell = StatusDialogUtil.getStatusShell();
-		assertEquals("Dialog is not centered correctly",getInitialLocation(shell), shell.getLocation());
+
+		Point expected = getInitialLocation(shell);
+		Point actual = shell.getLocation();
+		int tolerance = 1;
+		assertTrue("Dialog is not centered correctly" //
+				+ " (expected location: " + expected + " ± " + tolerance //
+				+ ", actual location: " + actual //
+				+ ", dialog bounds: " + shell.getBounds() //
+				+ ", parent bounds: " + shell.getParent().getBounds() + ")",
+				Math.abs(expected.x - actual.x) <= tolerance && Math.abs(expected.y - actual.y) <= tolerance);
 	}
 
 	//error link present

